src/library/commissioner_impl.cpp

```diff
@@ -495,15 +473,10 @@ void CommissionerImpl::GetRawActiveDataset(Handler<ByteArray> aHandler, uint16_t
     }
 #endif
 
-    if (mMeshLocalPrefix.empty())
-    {
-        // Request Active Dataset from the BA when we don't possess the mesh-local prefix.
-        mBrClient.SendRequest(request, onResponse);
-    }
-    else
-    {
-        mProxyClient.SendRequest(request, onResponse, GetLeaderLocator(), kDefaultMmPort);
-    }
+    // Always request Active Dataset from the Border Agent because
+    // the Mesh-Local prefix of current network is agnostic before
+    // we requesting the latest Active Operational Dataset.
```

Comment:
Can you try to reword this? I don't quite understand the "agnostic before we requesting" wording.

Reply:
Updated, please review.